src/otaclient/create_standby/update_slot.py

```diff
@@ -59,48 +59,64 @@ def __init__(
         self._fst_db_helper = file_table_db_helper
         self._status_report_queue = status_report_queue
         self.session_id = session_id
-        self._que: Queue[tuple[bytes, list[RegularFileTypedDict]] | None] = Queue()
+
+        # for process_regular workers
+        self._internal_que: Queue[int | None] = Queue()
 
         self._standby_slot_mp = Path(standby_slot_mount_point)
         self._resource_dir = Path(resource_dir)
 
         self.max_workers = max_workers
         self._se = threading.Semaphore(concurrent_tasks)
         self._interrupted = threading.Event()
-        self._thread_local = threading.local()
 
         self._hardlink_group = HardlinkGroup()
 
-    def _worker_initializer(self):
-        self._thread_local.merged_payload = UpdateProgressReport(
+    def _report_uploader_thread(self) -> None:
+        """Report uploader worker thread entry."""
+        _merged_report = UpdateProgressReport(
```

Comment:
Use a dedicated background thread to first gathering reports from each worker thread, and then pack them into one report to send to the status report que.